docs/user-guide/metadata.md

```diff
@@ -312,3 +315,69 @@ jobs:
 ```
 In the above example a Slack notification message will be send in `minimized` format for the `component` job if it was triggered by the scheduler.
 
+### Using [Lua](https://www.lua.org/) for atomic updates
+
+the `meta` tool creates a lock file, and holds a [flock](https://linux.die.net/man/2/flock) around each of its
+operations so that it may be executed by parallel operations in your build (for instance a Makefile invoked with
+`make -j 4`)
+
+In addition to atomic get and set operations, should "update" be required, the meta commands may be invoked by an
+embedded [Lua](https://www.lua.org/) interpreter.
+
+Example use cases:
+
+#### atomically increment a number
+
+```bash
+meta lua -E 'meta.set("myNum", (tonumber(meta.get("myNum")) or 0) + 1)'
+```
+
+| Lua code | explanation |
+| ---  | ---  |
+| `meta.get("myNum")` | gets the previous value |
+| `tonumber(`<small>meta.get("myNum")</small>`)` | `tonumber` returns its arg if number, parses if string, and returns `nil` when unparseable or not a number or string. |
+| <small>tonumber(meta.get("myNum"))</small>` or 0` | `or 0` Converts non-numbers to `0` so that arithmetic can be applied |
+| <small>(tonumber(meta.get("myNum")) or 0)</small>` + 1` | `+ 1` of course this increments the previous value (normalized to `0`) |
+| `meta.set(`<small>"myNum", (tonumber(meta.get("myNum")) or 0) + 1</small>`)` | `meta.set` sets the value to the incremented value |
```

Comment:
I'm kind of confused about the use of "`" in these lines

Reply:
Well… I was trying to come up with a way to highlight the progressive wrapping taking place - de-emphasizing the previous line with `<small>` tag, and emphasizing the new with backticks rather than `*` or `_` which both amount to just italicizing, sadly and aren't made _more_ prominent or as prominent as I wanted.

Reply:
Here's what it looks like with jekyll

![image](https://user-images.githubusercontent.com/41922797/104238064-ea786980-540d-11eb-871d-dfa599637e5f.png)